packages/site/src/components/TokensView.js

```diff
@@ -108,17 +111,53 @@ export const TokensView = () => {
 
 		setCoreColorTokens(sortedColorTokens);
 		setTokens(sortedTokens);
+
+		const initialDefaultTokens = ( async (e) => {
+			if ( 'default' === brand ) {
+
+				// Deep clone the object since we are checking equivalence later.
+				return JSON.parse( JSON.stringify( sortedTokens ) );
+			} else {
+				let url = TOKENS_FETCH_URL + "/default.raw.json";
+				let response = await fetch(url);
+				let json = await response.json();
+				let tokens = await json.props;
+
+				return tokens;
+			}
+		})();
+
+		// Save unchanged tokens so we can tell what is
+		// different from the original
+		setDefaultTokens( await initialDefaultTokens);
 	};
 
+
 	/**
 	 * Save JSON to file for browsers that support it, and fallback to
 	 * copy to clipboard for browsers that don't.
 	 *
 	 * @see https://developer.mozilla.org/en-US/docs/Web/API/window/showSaveFilePicker
 	 */
 	const saveJsonToFile = async () => {
+
+		const diffedTokens = ( ( changed, reference ) => {
+			const updatedTokenKeys = Object.keys( changed ).filter( key => {
+				return changed[key].value !== reference[key].value;
```

Comment:
This introduces a TypeError when updating an existing brand's tokens. Looks like `reference[key]` is undefined in that case.

Reply:
Good catch! This was happening when there are tokens in the changed object that are not in the reference object - i.e. theme-specific tokens.